src/coreclr/vm/ceemain.cpp

```diff
@@ -1835,12 +1835,20 @@ BOOL STDMETHODCALLTYPE EEDllMain( // TRUE on success, FALSE on error.
 
 struct TlsDestructionMonitor
 {
+    Thread* m_thread = nullptr;
+
+    void SetThread(Thread* thread)
+    {
+        _ASSERTE(thread == GetThreadNULLOk());
+        m_thread = thread;
+    }
+
     ~TlsDestructionMonitor()
     {
         // Don't destroy threads here if we're in shutdown (shutdown will
         // clean up for us instead).
 
-        Thread* thread = GetThreadNULLOk();
+        Thread* thread = m_thread;
```

Comment:
Should we delete the `ThreadDetach` call from `DeleteThread` and always depend on this path to cleanup the thread?

It would improve the test coverage for the thread cleanup logic - there won't be difference between thread cleanup for threads created by the runtime and threads that wandered into the runtime.

Reply:
Sounds reasonable.